torchinductor/dependencies.py

```diff
@@ -133,7 +136,15 @@ def canonicalize(self, index):
         # convert it to the simpliest form because of the interference from
         # different indexing formulas.
         index_vars = list(self._var_ranges.keys())
-        new_sizes, reindex, prune = _simplify_loops(index_vars, sizes, [index])
+        new_sizes, reindex, prune = _simplify_loops(
+            index_vars,
+            sizes,
+            [
+                index,
+                # added contiguous index prevents reordering
+                sympy_dot(index_vars, FlexibleLayout.contiguous_strides(sizes)),
```

Comment:
Rather than doing this in 3 different places let's refactor it into an option.

Reply:
Refactored